pkg/auth/auth.go

```diff
@@ -28,18 +29,66 @@ import (
 	yamlUtils "github.com/kubeapps/kubeapps/pkg/yaml"
 )
 
-// UserAuth contains information to check user permissions
-type UserAuth struct {
-	authCli      authorizationv1.AuthorizationV1Interface
-	discoveryCli discovery.DiscoveryInterface
-}
-
 type resource struct {
 	APIVersion string
 	Kind       string
 	Namespace  string
 }
 
+type k8sAuthInterface interface {
```

Comment:
why the `k8s` prefix? Couldn't userAuthInterface work? We don't plan on integrating with other auth mechanisms?

Reply:
there is not a good reason, I am bad at naming things :) I prefixed the `k8s` because these are the methods that contacts the K8s API